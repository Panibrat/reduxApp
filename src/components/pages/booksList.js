import React from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {getBooks} from '../../actions/booksActions';
import { Button, Grid, Row, Col } from 'react-bootstrap';
import BookItem from './BookItem';
import BooksForm from './booksForm';
import Cart from './cart';
import Time from './Time';

export class BooksList extends React.Component {
  componentDidMount() {
    this.props.getBooks();
    //console.log('componentDidMount');
  }
  componentDidUpdate() {
    //console.log('componentDidUpdate');
  }
  render() {
    const booksList = this.props.books.map((arrayBook) => {
      return (
        <Col xs={12} sm={6} md={4} key={arrayBook._id}>
          <BookItem
              _id={arrayBook._id}
              title={arrayBook.title}
              description={arrayBook.description}
              price={arrayBook.price}
          />
        </Col>
      )
    })

    return (
      <Grid style={{marginTop:'50px'}}>
        <Row>
          <Col >
            <Cart />
          </Col>
        </Row>
        <Row>
          <Time />
          <h1> Books List </h1>
          <Col xs={12} sm={6} md={4}>
            <BooksForm/>
          </Col>
          {booksList}
        </Row>
      </Grid>
    )
  }
}

function mapStateToProps(state){
  return {
    books: state.books.books
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    getBooks: getBooks
  }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(BooksList);
