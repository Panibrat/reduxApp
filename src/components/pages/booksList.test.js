import React from 'react';
import {shallow} from 'enzyme';
import {BooksList} from './booksList';

const books = [
  {
    _id: 1,
    title: 'First Book',
    description: 'this is the book description ',
    price: 10.25
  }, {
    _id: 2,
    title: 'Second',
    description: 'Description for Second',
    price: 22.88
  }, {
    _id: 3,
    title: 'Learn React',
    description: '4424h and you winner!!!',
    price: 300.30
  }
]


describe('Books List Snapshorts Rendering', () => {

  test('should render BooksList correctly with books in props', () => {
    const wrapper = shallow(<BooksList books={books} getBooks={()=>{}}/>);
    expect(wrapper).toMatchSnapshot();

  });

  test('should render empty BooksList', () => {
    const wrapper = shallow(<BooksList books={[]} getBooks={()=>{}}/>);
    expect(wrapper).toMatchSnapshot();

  });



});
