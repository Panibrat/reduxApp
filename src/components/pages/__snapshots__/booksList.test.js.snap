// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books List Snapshorts Rendering should render BooksList correctly with books in props 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}
  style={
    Object {
      "marginTop": "50px",
    }
  }
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
    >
      <Connect(Cart) />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Time />
    <h1>
       Books List 
    </h1>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      sm={6}
      xs={12}
    >
      <Connect(BooksForm) />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      key="1"
      md={4}
      sm={6}
      xs={12}
    >
      <Connect(BookItem)
        _id={1}
        description="this is the book description "
        price={10.25}
        title="First Book"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      key="2"
      md={4}
      sm={6}
      xs={12}
    >
      <Connect(BookItem)
        _id={2}
        description="Description for Second"
        price={22.88}
        title="Second"
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      key="3"
      md={4}
      sm={6}
      xs={12}
    >
      <Connect(BookItem)
        _id={3}
        description="4424h and you winner!!!"
        price={300.3}
        title="Learn React"
      />
    </Col>
  </Row>
</Grid>
`;

exports[`Books List Snapshorts Rendering should render empty BooksList 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}
  style={
    Object {
      "marginTop": "50px",
    }
  }
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
    >
      <Connect(Cart) />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Time />
    <h1>
       Books List 
    </h1>
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      sm={6}
      xs={12}
    >
      <Connect(BooksForm) />
    </Col>
  </Row>
</Grid>
`;
